### ISSUE TITLE
Wrong expected type in numpy format

### ISSUE THREAD

**Issue by steven-murray** [terrencepreilly/darglint#112]
In the following signature (using numpy mode), I get the error `src/edges_cal/cal_coefficients.py:from_load_name:609: DAR103:  ~load_name: expected str but was int`

Is this an instability in the numpy format?

```
@classmethod
    def from_load_name(
        cls,
        load_name: str,
        direc: [str, Path],
        run_num: Optional[int] = None,
        filetype: Optional[str] = None,
        **kwargs,
    ):
        """Instantiate the class from a given load name and directory.

        Parameters
        ----------
        load_name : str
            The load name (one of 'ambient', 'hot_load', 'open' or 'short').
        direc : str or Path
            The top-level calibration observation directory.
        run_num : int
            The run number to use for the spectra.
        filetype : str
            The filetype to look for (acq or h5).
        kwargs :
            All other arguments to :class:`LoadSpectrum`.

        Returns
        -------
        :class:`LoadSpectrum`.
        """
```

**Comment by steven-murray** [comment]
To further this issue, for this signature: 

```
    def __init__(
        self,
        spectrum: LoadSpectrum,
        reflections: SwitchCorrection,
        hot_load_correction: [HotLoadCorrection, None] = None,
        ambient: [LoadSpectrum, None] = None,
    ):
        """Wrapper class containing all relevant information for a given load.

        Parameters
        ----------
        spectrum : :class:`LoadSpectrum`
            The spectrum for this particular load.
```

I got the error `src/edges_cal/cal_coefficients.py:__init__:1046: DAR103:  ~spectrum: expected LoadSpectrum but was class`

**Comment by terrencepreilly** [comment]
There are some stability issues with Numpy format yet (see Issue #69).  I'll take a look.

**Comment by terrencepreilly** [comment]
So, there are a couple of things going on here.  The bug itself is occurring because there is an untyped argument, `kwargs`.  When darglint goes to verify types, it just zips arguments and types (a little naive, but it works most of the time.)  I'll have to fix that.  A momentary workaround is to specify `Dict[str, Any]` as the type of `kwargs`.

Another problem is that when darglint matches types, it will only accept the format given in the type signature.  That is, it will only accept something like `Union[str, Path]`, not `str or Path`.

**Comment by steven-murray** [comment]
I understand (kind of) the reason for allowing only `Union[str, Path]` rather than `str or Path`, but it strikes me as kind of redundant (at least when thinking about Sphinx docs), since the docs have the signature itself in them. It seems to me like the numpy standard was going for something closer to human readable, which would make it not quite as redundant. 

I guess if `darglint` would allow to have no types, we could choose to just leave the type in the signature itself.

**Comment by terrencepreilly** [comment]
Yes, you can leave types out, and darglint won't complain.  (You'll just have to drop the ":"s, as those should only be present with types.)  The point of including types was for when you convert a program from not using type hints (but having them present in the docstring) to using type hints.  It's just a reminder to either remove them, or have them match.

**Comment by steven-murray** [comment]
Perfect. Thanks for the explanation!

**Comment by terrencepreilly** [comment]
This fix should be present in v1.5.2.

**Comment by thomasgilgenast** [comment]
> Yes, you can leave types out, and darglint won't complain. (You'll just have to drop the ":"s, as those should only be present with types.)

Is this still the intended behavior @terrencepreilly? When I upgrade to 1.5.2 and with `docstring_style = numpy`, parameters in my docstrings without colons and without types all throw

```
DAR103 Parameter type mismatch:  ~<param_name>: expected <param_type> but was
```

Downgrading to 1.5.1 solves the issue.
